src/frontend/src/optimizer/plan_node/logical_except.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use itertools::Itertools;
+use risingwave_common::error::Result;
+
+use super::{ColPrunable, ExprRewritable, PlanBase, PlanRef, PredicatePushdown, ToBatch, ToStream};
+use crate::optimizer::plan_node::{
+    generic, ColumnPruningContext, LogicalFilter, PlanTreeNode, PredicatePushdownContext,
+    RewriteStreamContext, ToStreamContext,
+};
+use crate::utils::{ColIndexMapping, Condition};
+
+/// `LogicalExcept` returns the rows of its first input except any
+///  matching rows from its other inputs.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct LogicalExcept {
+    pub base: PlanBase,
+    core: generic::Except<PlanRef>,
+}
+
+impl LogicalExcept {
+    pub fn new(all: bool, inputs: Vec<PlanRef>) -> Self {
+        let core = generic::Except { all, inputs };
+        let base = PlanBase::new_logical_with_core(&core);
+        LogicalExcept { base, core }
+    }
```

Comment:
Does `Except` require all inputs have the same schema? If so, maybe it's better to have some assertion here and check the validity before creating `LogicalExcept`?

Reply:
Agreed. Currently, it's checked during binding. https://github.com/risingwavelabs/risingwave/blob/c86f2f5d81e0c8162f2c9b38f759c69e4e5094c5/src/frontend/src/binder/set_expr.rs#L143-L168